spec/integration/puppet/provider/powershell_spec.rb

```diff
@@ -0,0 +1,80 @@
+#! /usr/bin/env ruby
+require 'spec_helper'
+require 'puppet/type'
+require 'puppet/type/base_dsc'
+require 'fileutils'
+require 'puppet/file_system'
+
+describe Puppet::Type.type(:base_dsc).provider(:powershell), :if => Puppet.features.microsoft_windows? do
+
+  # Override initialize so we don't call the symlink set
+  class Puppet::Type::Base_dsc::ProviderPowershell
+    def initialize(value={})
+      super(value)
+    end
+  end
+
+  let (:resource) { Puppet::Type.type(:base_dsc).new(:provider => :powershell, :name => "windows_provider") }
+  let (:provider) { resource.provider }
+
+  let (:top_level_path) do
+    Dir.mktmpdir('dsc_playground')
+  end
+  let (:path) { top_level_path }
+  let (:program_files) { "#{path}\\Program Files"}
+  let (:vendored_modules_symlink_path){ "#{program_files}\\WindowsPowerShell\\Modules\\PuppetVendoredModules" }
+
+  context ".set_symlink_folder" do
+    before :each do
+      Win32::Registry.any_instance.expects(:[]).with('ProgramFilesDir').returns(program_files).at_most(10)
+      Puppet::FileSystem.unlink(vendored_modules_symlink_path) if Puppet::FileSystem.symlink?(vendored_modules_symlink_path)
+      system("rmdir /s /q \"#{program_files}\"") if ::File.directory?(program_files)
+      #FileUtils.rm_r(program_files) if ::File.directory?(program_files)
```

Comment:
Either way, this may need a pure verification. The symlinks are created calling `Puppet::FileSystem.symlink`, which calls `CreateSymbolicLinkW` - https://github.com/puppetlabs/puppet/blob/b3d9c03004bcb09d2539d48db666a2921f9bc481/lib/puppet/util/windows/file.rb#L82-L90